internal/constants/scripts_test.go

```diff
@@ -0,0 +1,170 @@
+//nolint:gosec
+package constants
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Pre Network Config Script", func() {
+	type replace struct {
+		from string
+		to   string
+	}
+	type expectedFile struct {
+		source      string
+		destination string
+		replaces    []replace
+	}
+	const assistedNetwork = "test_root/etc/assisted/network"
+	var (
+		root              string
+		cmd               string
+		err               error
+		systemConnections string
+	)
+	BeforeEach(func() {
+		root, err = os.MkdirTemp("", "test_root")
+		Expect(err).ToNot(HaveOccurred())
+		var f *os.File
+		f, err = os.CreateTemp("", "script.sh")
+		Expect(err).ToNot(HaveOccurred())
+		var n int
+		n, err = f.WriteString(PreNetworkConfigScript)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(n).To(Equal(len(PreNetworkConfigScript)))
+		cmd = f.Name()
+		Expect(f.Chmod(0o755)).ToNot(HaveOccurred())
+		f.Close()
+		systemConnections = root + "/etc/NetworkManager/system-connections"
+	})
+	AfterEach(func() {
+		Expect(os.RemoveAll(root)).ToNot(HaveOccurred())
+		Expect(os.RemoveAll(cmd)).ToNot(HaveOccurred())
+	})
+	copyFiles := func() {
+		_, err = exec.Command("bash", "-c", "cp -r test_root/* "+root).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+	copySys := func() {
+		_, err = exec.Command("bash", "-c", "cp -r test_root/sys "+root).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+	copyUnmatchingDir := func() {
+		dir := root + "/etc/assisted/network"
+		Expect(os.MkdirAll(dir, 0o777)).ToNot(HaveOccurred())
+		_, err = exec.Command("bash", "-c", "cp -r test_root/etc/assisted/network/host1 "+dir).CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+	verifyExpectedFiles := func(expectedFiles []expectedFile) {
+		for _, e := range expectedFiles {
+			src, err := os.ReadFile(e.source)
+			Expect(err).ToNot(HaveOccurred())
+			dst, err := os.ReadFile(e.destination)
+			Expect(err).ToNot(HaveOccurred())
+			source := string(src)
+			destination := string(dst)
+			if len(e.replaces) > 0 {
+				Expect(source).ToNot(Equal(destination))
+			}
+			for _, r := range e.replaces {
+				source = strings.ReplaceAll(source, r.from, r.to)
+			}
+			Expect(source).To(Equal(destination))
+		}
+	}
+	It("No host directory", func() {
```

Comment:
Can you name tests with the context and expected result?

Something like:

```suggestion
	It("script runs successfully when there are no host directories", func() {
```

This is also something that should never happen as we only add the script if we have static networking, but still reasonable to test.